### PR TITLE
📦 Implement Laravel 6 compatibility

### PR DESCRIPTION
**Pull request by rdarcy1**
Currently upgrading a project to Laravel 6 and saw this package needed a few tweaks. I think the only actual changes required in the src directory are removing the global helper functions, but to get it working with the L6 testbench version (4.x) it needs PHPUnit 8 and all the related changes there.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -4,13 +4,13 @@
...
-    "illuminate/support": "^5.4",
-    "illuminate/database": "^5.4",
+    "illuminate/support": "^5.4 | ^6.0",
+    "illuminate/database": "^5.4 | ^6.0",
     "ext-json": "*"
   },
   "require-dev": {
-    "phpunit/phpunit": "~5.7",
-    "orchestra/testbench": "~3.0",
+    "phpunit/phpunit": "^8",
```
Comment (mathieutu) [inline composer.json]:
```suggestion
    "phpunit/phpunit": "^8.0",
```

File: phpunit.xml [hunk 2]
```diff
@@ -9,7 +9,6 @@
          processIsolation="false"
          stopOnError="false"
          stopOnFailure="false"
-         syntaxCheck="true"
```
Comment (mathieutu) [inline phpunit.xml:12]: Could you explain this?

**Review by mathieutu — APPROVED** [review]
Thank you for contribution!
The tests are failing for php 7.0 and 7.1, but actually they are not maintained anymore, so we can remove the support for them in this release.

The only thing I see is that we only test the Laravel 6 version, so we can't see if there is a breaking change with previous ones.

Do you mind updating the travis.yml file with a matrix and the supported version of Laravel? An explanation can be found here: https://www.tjmiller.me/blog/verifying-laravel-version-compatibility/ and a simple example here: https://github.com/spatie/laravel-view-models/blob/master/.travis.yml

Thank you again.

TL;DR:

- [ ] Remove support for php < 7.2 (compose.json and travis.yml)
- [ ] Add a Laravel version Matrix in tests.

**Comment by rdarcy1** [comment]
No problem, thanks for the package! I'll have a look at those changes.

File: phpunit.xml (hunk 2 above)
Comment (rdarcy1) [reply phpunit.xml:12]: In PHPUnit 8 it produces a warning as an unrecognised parameter – I think in previous versions it was just ignored (but still did nothing). 

https://stackoverflow.com/questions/44328114/phpunit-what-does-syntaxcheck-configuration-parameter-stands-for-exactly/44331140#44331140

Comment (mathieutu) [reply phpunit.xml:12]: Ok thanks!

**Comment by rdarcy1** [comment]
I've added a version matrix – hopefully it's correct! Not sure what's causing the Travis failures?

```
Source directory /home/travis/build/mathieutu/laravel-json-syncer/vendor/nesbot/carbon has uncommitted changes.       
```

**Comment by mathieutu** [comment]
Thanks, it's perfect!
Don't worry I'll take over next.
